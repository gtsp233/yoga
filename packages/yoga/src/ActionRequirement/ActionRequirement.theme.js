const ActionRequirement = () => ({});

export default ActionRequirement;
