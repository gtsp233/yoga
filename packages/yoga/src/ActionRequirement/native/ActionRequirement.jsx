import React from 'react';
import { arrayOf, node, oneOfType, string } from 'prop-types';
import styled from 'styled-components/native';
import { View } from 'react-native';
import Box from '../../Box';
import { Title } from './ActionRequirementStyles';
import Text from '../../Text';

const StyledActionRequirement = styled.View``;

function ActionRequirement(props) {
  const { title, description, children, checkable, illustration, list } = props;

  return (
    <StyledActionRequirement {...props}>
      {illustration && <Box>{illustration}</Box>}
      <Title>{title}</Title>
      <Text mt="small" color="deep">
        {description}
      </Text>
      <Text mt="small" color="deep">
        {list && <Box mt="xxxlarge">{list}</Box>}
      </Text>
      {checkable && <Box mt="xxxlarge">{checkable}</Box>}
      <View>{children}</View>
    </StyledActionRequirement>
  );
}

ActionRequirement.propTypes = {
  title: string.isRequired,
  children: oneOfType([arrayOf(node), node]),
  description: string.isRequired,
  checkable: oneOfType([arrayOf(node), node]),
  illustration: oneOfType([arrayOf(node), node]),
  list: oneOfType([arrayOf(node), node]),
};

ActionRequirement.defaultProps = {
  children: undefined,
  checkable: undefined,
  illustration: undefined,
  list: undefined,
};

export default ActionRequirement;
