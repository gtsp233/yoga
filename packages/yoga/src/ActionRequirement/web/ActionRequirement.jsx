import React from 'react';
import { arrayOf, node, oneOfType, string } from 'prop-types';
import styled from 'styled-components';
import {
  Actions,
  Title,
  PrimaryButton,
  SecondaryButton,
} from './ActionRequirementStyles';
import Text from '../../Text';
import Box from '../../Box';

const StyledActionRequirement = styled.div``;

function isChildFromComponent(child, component) {
  return child.type.displayName === component.displayName;
}

function ActionRequirement(props) {
  const { title, description, children, checkable, illustration, list } = props;

  let primaryButton;
  let secondaryButton;

  function defineCompoundComponents() {
    React.Children.forEach(children, child => {
      if (isChildFromComponent(child, PrimaryButton)) primaryButton = child;
      if (isChildFromComponent(child, SecondaryButton)) secondaryButton = child;
    });
  }

  defineCompoundComponents();
  return (
    <StyledActionRequirement {...props}>
      {illustration && <Box>{illustration}</Box>}
      <Title>{title}</Title>
      <Text mt="small" color="deep">
        {description}
      </Text>
      <Text mt="small" color="deep">
        {list && <Box mt="xxxlarge">{list}</Box>}
      </Text>
      {checkable && <Box mt="xxxlarge">{checkable}</Box>}
      <Actions mt="xlarge">
        {primaryButton}
        {secondaryButton}
      </Actions>
    </StyledActionRequirement>
  );
}

ActionRequirement.propTypes = {
  title: string.isRequired,
  children: oneOfType([arrayOf(node), node]),
  description: string.isRequired,
  checkable: oneOfType([arrayOf(node), node]),
  illustration: oneOfType([arrayOf(node), node]),
  list: oneOfType([arrayOf(node), node]),
};

ActionRequirement.defaultProps = {
  children: undefined,
  checkable: undefined,
  illustration: undefined,
  list: undefined,
};

export default ActionRequirement;
