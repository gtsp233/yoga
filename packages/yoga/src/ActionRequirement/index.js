import ActionRequirement from './web';

export default ActionRequirement;
