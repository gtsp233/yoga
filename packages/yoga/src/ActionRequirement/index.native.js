import ActionRequirement from './native';

export default ActionRequirement;
