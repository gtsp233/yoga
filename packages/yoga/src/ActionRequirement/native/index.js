import ActionRequirement from './ActionRequirement';
import {
  Title,
  PrimaryButton,
  SecondaryButton,
} from './ActionRequirementStyles';

ActionRequirement.Title = Title;
ActionRequirement.PrimaryButton = PrimaryButton;
ActionRequirement.SecondaryButton = SecondaryButton;

export default ActionRequirement;
