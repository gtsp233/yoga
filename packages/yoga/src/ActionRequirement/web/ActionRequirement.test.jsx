import React from 'react';
import { render } from '@testing-library/react';

import { ThemeProvider } from '../..';
import ActionRequirement from './ActionRequirement';

describe('<ActionRequirement />', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider>
        <ActionRequirement />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
