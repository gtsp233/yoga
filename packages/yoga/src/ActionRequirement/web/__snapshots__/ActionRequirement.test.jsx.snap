// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionRequirement /> should match snapshot 1`] = `
.c1 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 24px;
  font-weight: 400;
  font-family: Rubik;
  color: #231B22;
  margin-top: 16px;
  color: #6B6B78;
}

.c0 {
  margin: 0;
  padding: 0;
  font-size: 24px;
  line-height: 32px;
  font-weight: 500;
  font-family: Rubik;
  color: #231B22;
}

.c2 {
  margin-top: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media (min-width:769px) {
  .c0 {
    max-width: 700px;
    font-size: 48px;
    line-height: 48px;
  }
}

@media (min-width:769px) {
  .c2 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    margin-top: 32px;
  }
}

<div>
  <div
    class=""
  >
    <p
      class="c0"
    />
    <p
      class="c1"
      color="deep"
    />
    <p
      class="c1"
      color="deep"
    />
    <div
      class="c2"
    />
  </div>
</div>
`;
